### PR TITLE
[CBR-423] Structured logging of data structures

### PR DESCRIPTION
**Pull request by 365andreas**
## Description

Structured logging of data structures.

Up to the current version of code only Text  messages are logged. So we want to log items (besides Text ) that have a ToJSON  instance.

## Linked issue

[CBR-423](https://iohk.myjetbrains.com/youtrack/issue/CBR-423)

## Type of change

- [~] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🛠 New feature (non-breaking change which adds functionality)
- [~] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [~] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [~] 🔨 New or improved tests for existing code
- [~] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [~] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [~] I have added tests to cover my changes.
- [~] All new and existing tests passed.

### CODE REVIEW

File: log-configs/template-demo.yaml [hunk 1]
```diff
@@ -8,3 +8,9 @@ loggerTree:
   files:
     - {{file}}.pub
```
Comment (CodiePP) [inline log-configs/template-demo.yaml]: we can actually remove the *.pub file as the JSON log will replace it

Comment (365andreas) [reply log-configs/template-demo.yaml]: Done.

**Comment by 365andreas** [comment]
@erikd one last commit added and rebased again

**Comment by CodiePP** [comment]
* a new demo structure in the test `Test.Pos.Util.LogStructuredSpec`
* removed `ToJSON` instances of LoggerConfig (not used anywhere)
* rebased on `develop`

**Pull request merged**